src/deepsparse/benchmark/benchmark_sweep.py

```diff
@@ -102,49 +167,223 @@ def benchmark_sweep(
                     ]
+                    command_str = f"\"{' '.join(command)}\""
+
+                    writer.writerow(
+                        [
+                            model,
+                            engine,
+                            batch_size,
+                            num_core,
+                            scenario,
+                            num_streams,
+                            items_per_second,
+                            latency_mean,
+                            latency_median,
+                            latency_std,
+                            input_shape,
+                            command_str,
+                        ]
+                    )
+                    _LOGGER.info(
+                        f"{model} benchmarking results written to {export_csv_path}"
+                    )
+                except Exception as exception:
+                    _LOGGER.info(
+                        f"An exception was raised while trying to benchmark {model}, "
+                        f"with batch_size: {batch_size}, input shapes: {input_shape},"
+                        f"num_cores: {num_core} and num_streams: {num_streams}, "
+                        f"Exception: {exception}"
+                    )
+
+
+def _get_models(
+    model_dir: Optional[str] = None,
+    model_paths: Optional[str] = None,
+) -> List[str]:
+    models = []
+
+    if model_dir is not None:
+        models.extend([str(model) for model in Path(model_dir).rglob("*.onnx")])
+
+    if model_paths is not None:
+        more_models = model_paths.split(",")
+        for model in more_models:
+            model = model.strip()
+
+            if Path(model).exists() or model.startswith("zoo:"):
+                models.append(model)
+            else:
+                raise ValueError(
+                    "The specified models must either be valid paths that exist,"
+                    f"or a valid SparseZoo stub but found {model}"
                 )
+    if not models:
+        raise ValueError(
+            "Could not find any models, either `--model-dir` did not have `onnx` files "
+            "or was not specified, additionally `--model-paths` was also not specified"
+        )
+    return models
+
+
+@click.command()
+@click.option(
+    "--num-cores",
+    help="Comma separated values for different num_cores to benchmark against, "
+    "can also be specified as max to benchmark against max num of cores",
+)
+@click.option(
+    "--batch-sizes",
+    help="Comma separated values for different batch_sizes to benchmark against",
+)
+@click.option(
+    "--model-dir",
+    type=str,
+    help="Directory of model files to benchmark, if not specified then "
+    "`--model-paths` must be specified",
+    default=None,
+)
+@click.option(
+    "--model-paths",
+    type=str,
+    default=None,
+    help="Comma separated list of model paths, or Sparsezoo stubs. Must be specified"
+    " if no `--model-dir` given",
+)
```

Comment:
why not just `--models`?

load with `.split(",")`, and unwrap any paths that are dirs

Reply:
great callout, update coming up soon!

Reply:
Done! kept the arg name same though